app/scripts/lib/validate.js

```diff
@@ -13,6 +13,38 @@ define(function (require, exports, module) {
   // taken from the fxa-auth-server
   var HEX_STRING = /^(?:[a-fA-F0-9]{2})+$/;
 
+  // URL RegEx taken from http://blog.mattheworiordan.com/post/13174566389/url-regular-expression-for-links-with-or-without
+  // jscs:disable maximumLineLength
+  var urlRegEx = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)/;
+  // jscs:enable maximumLineLength
+
+  // URN Regex
+  var urnRegEx = /^urn:[a-zA-Z0-9][a-zA-Z0-9-]{1,31}:([a-zA-Z0-9()+,.:=@;$_!*'-]|%[0-9A-Fa-f]{2})+$/;
+
+  /**
+   * Check whether string is a url. Validates url with or
+   * without http(s) and trailing slash.
+   *
+   * http://blog.mattheworiordan.com/post/13174566389/url-regular-expression-for-links-with-or-without
+   */
+  function isUrlValid (url) {
+    return urlRegEx.test(url);
```

Comment:
Yup, there is a reason I haven't used window.URL before, I just can't remember what it is.